### DISCUSSION TITLE
how can I reduce file fragmentation?

### DISCUSSION THREAD

**Discussion by rezad1393** [transmission/transmission#2217] [Q&A] [answered]
I have installed trasnmission on archlinux.
I have set allocation to 2 (full)
my filesystem is ext4.
when I download a big file (2-3GB) I get a lot of fragmentation by this command
find /mnt/downloads/finished -xdev -type f -exec filefrag {} +

and e4defrag shows those too (although with lower number of fragments which I don't know why)

how can I make this better (less fragments)?

**Comment by ckerr** [comment, marked as answer]
Using full preallocation should use `fallocate64()` on POSIX systems, so this should do the trick on ext4. This is the best approach that I'm aware of on modern systems. If you're still getting fragmentation, and you know of a better way to preallocate files, patches are welcomed, or if you don't want to write the patch, at least file an issue that documents a better approach than fallocate() / fallocate64() :smile:
